src/System.Linq/src/System/Linq/ElementAt.cs

```diff
@@ -47,14 +47,14 @@ public static TSource ElementAt<TSource>(this IEnumerable<TSource> source, int i
                 }
             }
 
-            throw Error.ArgumentOutOfRange(nameof(index));
+            throw ThrowHelper.CreateArgumentOutOfRangeException(ExceptionArgument.index);
```

Comment:
Why does this use `throw ThrowHelper.Create` and not `ThrowHelper.Throw` ?

Reply:
Because the csc flow analysis doesn't know that ThrowXx is non-returning, so I'd need to put a `return default` or something like that after it; this seemed cleaner.  You disagree?

Reply:
Ok, that's fine. It was just not obvious - my eye tends to be trained for the `return default;` for cases like this. 

Similar discussion https://github.com/dotnet/coreclr/pull/22321#discussion_r254452697

Reply:
`ThrowHelper.Throw` + `return default;` should generate slightly smaller code.

Reply:
Ok, I'll change them.